cli/tools/registry/pm.rs

```diff
@@ -463,6 +451,37 @@ pub async fn add(
   let jsr_resolver = Arc::new(JsrFetchResolver::new(deps_file_fetcher.clone()));
   let npm_resolver = Arc::new(NpmFetchResolver::new(deps_file_fetcher));
 
+  let mut selected_packages = Vec::with_capacity(add_flags.packages.len());
+  let mut package_reqs = Vec::with_capacity(add_flags.packages.len());
+
+  for entry_text in add_flags.packages.iter() {
+    let req = AddPackageReq::parse(entry_text).with_context(|| {
+      format!("Failed to parse package required: {}", entry_text)
+    })?;
+
+    match req {
+      Ok(add_req) => package_reqs.push(add_req),
+      Err(package_req) => {
+        if jsr_resolver.req_to_nv(&package_req).await.is_some() {
+          bail!(
+            "{entry_text} is missing a prefix. Did you mean `{}`?",
+            crate::colors::yellow(format!("deno {cmd_name} jsr:{package_req}"))
+          )
+        } else if npm_resolver.req_to_nv(&package_req).await.is_some() {
```

Comment:
I'm not sure if this is fully correct - what is a package is available on both? Maybe run these concurrently?